sw/host/opentitanlib/src/transport/cw310/mod.rs

```diff
@@ -60,13 +64,36 @@ impl CW310 {
     }
 
     // Initialize the IO direction of some basic pins on the board.
-    fn init_direction(&self) -> Result<()> {
+    fn init_direction(&self) -> anyhow::Result<()> {
         let device = self.device.borrow();
         device.pin_set_output(Self::PIN_SRST, true)?;
         device.pin_set_output(Self::PIN_JTAG, true)?;
         device.pin_set_output(Self::PIN_BOOTSTRAP, true)?;
         Ok(())
     }
+
+    fn open_uart(&self, instance: u32) -> Result<SerialPortUart> {
+        let usb = self.device.borrow();
+        let serial_number = usb.get_serial_number();
+
+        let mut ports = serialport::available_ports().wrap(UartError::EnumerationError)?;
+        ports.retain(|port| {
+            if let SerialPortType::UsbPort(info) = &port.port_type {
```

Comment:
Kinda wonder if this could be simplified using my `extrude` crate: https://github.com/mcy/extrude

Reply:
See comment beow.